woocommerce/woocommerce
# Add order attribution install banner in analytics overview's header

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of #51567, this PR adds the order attribution install banner in the core analytics overview page's header area. The banner will be shown if:

- The `woocommerce-analytics` plugin is not installed yet, and
- The merchant has clicked the `Dismiss` button on the main banner, which implemented in #51672.

https://github.com/user-attachments/assets/d7a1e2ec-68e4-47db-99b6-dcf6e4835d9f

Note that when clicking `Try Order Attribution` on the banner, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found as the plugin hasn't been published.

#### Technical details

Since the header banner is simply just a button, I did not use the shared banner implemented in the previous PR #51672.

#### Event tracking

Same as in #51672, this PR sends two events:
- `order_attribution_install_banner_viewed`
- `order_attribution_install_banner_clicked`

But with different `context` property: `analytics-overview-header`.

#### Next step
- Display the banner in other places within WooCommerce Admin
- Implement displaying the banner for % of merchants
- The changelog entry will be added in the final PR once all the banners are added



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the Console tab in the browser DevTool, and run `localStorage.setItem( 'debug', 'wc-admin:*' )` to enable tracking debugging mode
2. Do not install `woocommerce-analytics` plugin
3. Go to `Analytics -> Overview`
4. It should show the main banner
7. Click `Dismiss`
8. The main banner is gone
9. The small banner in the header should appear
10. The event `wcadmin_order_attribution_install_banner_viewed` with `context: analytics-overview-header` should be sent
11. Click `Try Order Attribution`, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found
12. The event `wcadmin_order_attribution_install_banner_clicked` with `context: analytics-overview-header` should be sent
15. Delete the option `woocommerce_order_attribution_install_banner_dismissed` from the DB
16. Go to `Analytics -> Overview`, the header banner should be shown again
17. Click `Dismiss` again on the main banner
18. Install `woocommerce-analytics` plugin, do not activate it
19. Go to `Analytics -> Overview`, the header banner should not be shown
20. Activate `woocommerce-analytics` plugin
21. Go to `Analytics -> Overview`, the header banner should not be shown 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
